### PR TITLE
First task complete.

### PR DESCRIPTION
**Pull request by martinmilev**
The first task ready for review.

### CODE REVIEW

File: src/ex1/Demo.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    String str = "Not to mention rumours which agitated the maritime population and excited the public mind.";
+    System.out.println("Enter Page size.");
+    Scanner sc = new Scanner(System.in);
+    int pageSize = sc.nextInt();
+    pb.add(str, pageSize);
+    System.out.println(pb.next());
+    System.out.println("\"x\" - exit,\"n\" - next, \"p\" - previous,\"f\" - first page, \"l\" - last page, \"#\" - page number.");
+    while (true) {
+      String s = sc.nextLine();
+      if(s.equals("n")) System.out.println(pb.next());
```
Comment (mgenov) [inline src/ex1/Demo.java]: brackets are missing

File: src/ex1/Demo.java [hunk 2]
```diff
@@ -0,0 +1,27 @@
+package ex1;
```
Comment (mgenov) [inline src/ex1/Demo.java]: please use better package names like: com.clouway.collections.pagebean

File: src/ex1/LastPageException.java [hunk 3]
```diff
@@ -0,0 +1,7 @@
+package ex1;
+
+public class LastPageException extends Exception {
```
Comment (mgenov) [inline src/ex1/LastPageException.java]: Doesn't sound like a good name for error. What is the "error" meaning of LastPage ?

File: src/ex1/Demo.java (hunk 2 above)
Comment (martinmilev) [inline src/ex1/Demo.java]: Done. For some reason src/ex1/ still exists in the origin. I will remove it on the next commit.

File: src/ex1/Demo.java (hunk 1 above)
Comment (martinmilev) [inline src/ex1/Demo.java]: Done.

File: src/ex1/LastPageException.java (hunk 3 above)
Comment (martinmilev) [inline src/ex1/LastPageException.java]: Done. Renamed both exceptions.

File: src/com/clouway/collections/pagebean/ExceptionNextPageNotExist.java [hunk 4]
```diff
@@ -0,0 +1,7 @@
+package com.clouway.collections.pagebean;
+
+public class ExceptionNextPageNotExist extends Exception {
```
Comment (mgenov) [inline src/com/clouway/collections/pagebean/ExceptionNextPageNotExist.java]: Compare your naming convention with these: http://docs.oracle.com/javase/7/docs/api/java/lang/Exception.html

Comment (mgenov) [inline src/com/clouway/collections/pagebean/ExceptionNextPageNotExist.java]: Point: Direct Known Subclasses

Comment (martinmilev) [inline src/com/clouway/collections/pagebean/ExceptionNextPageNotExist.java]: "It is good practice to add Exception to all exception classes. Also the name of the exception should be meaningful, should represent the problem. For example CustomerNotFoundException indicate that customer was not found."

So is NextPageNotFoundException okay for naming the exception?

Comment (mgenov) [inline src/com/clouway/collections/pagebean/ExceptionNextPageNotExist.java]: Yes, Sounds good.

Comment (martinmilev) [inline src/com/clouway/collections/pagebean/ExceptionNextPageNotExist.java]: WA

File: src/com/clouway/collections/sentencesprocessor/SentencesProcessor.java [hunk 5]
```diff
@@ -0,0 +1,68 @@
+package com.clouway.collections.sentencesprocessor;
+
+import java.util.Arrays;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.Collections;
+
+public class SentencesProcessor {
+  private ArrayList<String> list;
```
Comment (mgenov) [inline src/com/clouway/collections/sentencesprocessor/SentencesProcessor.java]: As we already discussed, using of parent interface is preferable.

File: src/com/clouway/collections/sentencesprocessor/SentencesProcessor.java [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+    return str;
+  }
+
+  /**
+   * Returns the most frequent letter in a sentence.
+   *
+   * @return
+   */
+  public String letterCount() {
+    String sentence = cleanup(list.toString());
```
Comment (mgenov) [inline src/com/clouway/collections/sentencesprocessor/SentencesProcessor.java]: Using of global list is not a good idea. list.toString() is not same as "somestring", cause it adds additional characters.

File: src/com/clouway/collections/sentencesprocessor/SentencesProcessor.java (hunk 5 above)
Comment (mgenov) [inline src/com/clouway/collections/sentencesprocessor/SentencesProcessor.java]: Actually I'm not sure that you need this list. The idea of this task is Map/Hashtable introduction.

**Comment by mgenov** [comment]
No LGTM ?

**Comment by martinmilev** [comment]
It was not my intention to close the pull request.

**Pull request closed without merging**